src/core/metadata.ts

```diff
@@ -12,3 +12,14 @@
  * this with different values in the future.
  */
 export type Metadata = Record<string, unknown>;
+
+/**
+ * The type for metadata contained in the details fields of a chat message.
+ * This is a key-value pair where the key is a string, and the value is a string, it represents the metadata supplied
+ * to a message update or deletion request.
+ *
+ * Do not use metadata for authoritative information. There is no server-side
+ * validation. When reading the metadata, treat it like user input.
+ *
+ */
+export type DetailsMetadata = Record<string, string>;
```

Comment:
I'm not sure about this living here - `Metadata` is related specifically to the message entity, whereas `DetailsMetadata` is about operations. Should they live together?

Reply:
That's a fair point, since they are both linked to the message entity, the latter via the new operation field, I put them together. I'll move the metadata into a new operations.ts file.